chaosk8s/pod/actions.py

```diff
@@ -128,7 +129,7 @@ def exec_in_pods(
         v1, label_selector, name_pattern, all, rand, mode, qty, ns, order
     )
 
-    exec_command = cmd.strip().split() if isinstance(cmd, str) else cmd
+    exec_command = shlex.split(cmd) if isinstance(cmd, str) else cmd
```

Comment:
Please add a specific test that was falling before and not falling after this fix.

Reply:
I am not sure where to add the test case. But error I was getting prior to this change is:
`[2022-04-01 12:49:47 DEBUG] [actions:166] Failed loading pod exec error stream as a json payload
    Traceback (most recent call last):
      File "/Users/plashgar/.venvs/chaostk/lib/python3.9/site-packages/chaosk8s/pod/actions.py", line 164, in exec_in_pods
        err = json.loads(err)
      File "/usr/local/Cellar/python@3.9/3.9.10/Frameworks/Python.framework/Versions/3.9/lib/python3.9/json/__init__.py", line 346, in loads
        return _default_decoder.decode(s)
      File "/usr/local/Cellar/python@3.9/3.9.10/Frameworks/Python.framework/Versions/3.9/lib/python3.9/json/decoder.py", line 337, in decode
        obj, end = self.raw_decode(s, idx=_w(s, 0).end())
      File "/usr/local/Cellar/python@3.9/3.9.10/Frameworks/Python.framework/Versions/3.9/lib/python3.9/json/decoder.py", line 355, in raw_decode
        raise JSONDecodeError("Expecting value", s, err.value) from None
    json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)`

where the experiment would then not run. After this fix, the experiment runs as normal with no errors.

Reply:
You should add a test at the end of the module https://github.com/chaostoolkit/chaostoolkit-kubernetes/blob/master/tests/test_pod.py

Try to follow the other tests to get an idea of how to mock the function.

Reply:
ah haven't worked with unittests in python before, but let me take a look and see. Thanks!

Reply:
I have added a test case, more like copied one over since I have no clue what is really happening in that file. But the main goal was for it to just check a shlex like command instead of string or a list? hopefully it's good enough.